src/schema/v2/object_identification.ts

```diff
@@ -158,6 +161,13 @@ const NodeField: GraphQLFieldConfig<any, ResolverContext> = {
       ).then((data) => {
         // Add the already known type so `NodeInterface` can pluck that out in
         // its `resolveType` implementation.
+        // If `type` is a union type, we need to return the more specific `type`
+        // based on the union type's `resolveType` and `data`
+        if (type.resolveType) {
+          const resolvedType = type.resolveType(data)
+          return { __type: resolvedType, ...data }
+        }
+
```

Comment:
**question (non-blocking):** Neat! I believe union type resolution comes in two flavors: `resolveType` vs `isTypeOf`. Is there anything to do here to support the latter kind?

Reply:
Nope! I think `resolveType` is what we want as that's declared on the actual union type - the `HomeViewSection` in this case - and returns a type. The `isTypeOf` returns a boolean and is what you might declare on the narrowed type if needed.

Reply:
Gotcha, I was just thinking there might be other unions that are defined using the other flavor ([example](https://github.com/artsy/metaphysics/blob/f57600cc020d44886e1be7846c87162dbb4e7311/src/schema/v2/article/index.ts#L282-L292) [here](https://github.com/search?q=repo%3Aartsy%2Fmetaphysics+path%3Asrc%2Fschema%2Fv2%2Farticle+isTypeOf&type=code)), and whether we wanted to preemptively support that since we are coming up with a general purpose solution here. But it may just work anyway if this [old comment](https://github.com/graphql/graphql-js/issues/876#issuecomment-304398882) is still accurate.

Reply:
Ah, tricksy! Yea, I would keep it with `resolveType` for now - this `Node` thing is very specific (you need to opt-in) and homegrown, so we can expand it when needed, vs. anything pre-emptive.

Reply:
er I guess said differently - you can use _either_ `resolveType` or `isTypeOf` for a union type + the specific types. This solution seems to work for `resolveType` - which is what we use for homeview.

For unions where `isTypeOf` pattern is used instead - unclear if this will work - _maaybe_ if `resolveType` is getting created/generated or something?

Either way, that feels like something to add when needed I suppose. In that case, this PR is more about supporting unions **that use `resolveType`**.